Sources/SwiftDocC/Infrastructure/Bundle Assets/SVGIDExtractor.swift

```diff
@@ -0,0 +1,57 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2022 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See https://swift.org/LICENSE.txt for license information
+ See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import Foundation
+
+// On non-Darwin platforms, Foundation's XML support is vended as a separate module:
+// https://github.com/apple/swift-corelibs-foundation/blob/main/Docs/ReleaseNotes_Swift5.md#dependency-management
+#if canImport(FoundationXML)
+import FoundationXML
+#endif
+
+/// A basic XML parser that extracts the first `id` attribute found in the given SVG.
+///
+/// This is a single-purpose tool and should not be used for general-purpose SVG parsing.
+enum SVGIDExtractor {
```

Comment:
In order to fully integrate with Swift-DocC-Render's theming system for custom icons, we need to be able to provide the `id` of any custom SVGs we're using as icons.

Asking the user to provide these would be unfortunate since it asks them to have a greater understanding of SVGs than they might have. So instead, this is a very basic SVG parser that just extracts the first id tag it finds. We're still asking the user to provide a valid SVG, but assuming they do provide a valid one, this will pull out the ID that DocC-Render needs.

Reply:
We should just make sure to point out the `<svg` element of the icon needs **some** id. They can add with any code/text editor

Reply:
Yeah definitely. Have this well documented will be really important.

Reply:
Do we still need a warning when there's no ID?

Reply:
Also, what will happen if a custom SVG icon doesn't have an ID?

Reply:
rdar://99205219 tracks supporting rendering SVGs without IDs and other image types PNGs. SVGs with IDs will always be the _best_ option since it allows DocC-Render to apply color tint and scaling more correctly but I'm still hoping we can provide a reasonable UX when those are not provided.

If we investigate more and that turns out to be unfeasible then we should add a warning here. So rdar://99205219 can track that eventuality as well if need be.

CC: @mportiz08